sympy/parsing/mathematica.py

```diff
@@ -1,133 +1,433 @@
 from __future__ import print_function, division
 
+from collections import defaultdict
+from itertools import product
 import re
 from sympy import sympify
 
 
-def mathematica(s):
-    return sympify(parse(s))
+class mathematica(object):
```

Comment:
Why did you make this a class?

Reply:
It is because I wanted the `mathematica` class to keep the `parse` instance. 
The `parse` instance has a cache for the latest users' `additional_translations` dictionary. The `additional_translations` dictionary needs to be transformed internally for make parsing easier, and this cache saves time from transforming it every time  when the `mathematica` is called.